### PR TITLE
[Io-759] Meta Api Client

### PR DESCRIPTION
**Pull request by Nathanjp91**
Creation of meta api client object for use with higher level and user centric interaction with darwin objects

### Changelog
Introduction of MetaClient object and associated functionality

### CODE REVIEW

File: darwin/future/core/client.py [hunk 1]
```diff
@@ -93,34 +124,6 @@ class Result(BaseModel):
     ...
 
 
-class PageDetail(BaseModel):
```
Comment (Nathanjp91) [inline darwin/future/core/client.py:96]: Don't need these if not using cursors, remove for cleaner codebase

File: darwin/future/meta/client.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    def __init__(self, config: DarwinConfig, retries: Optional[Retry] = None) -> None:
+        super().__init__(config, retries=retries)
+
+    @classmethod
+    def local(cls) -> Client:
+        config = DarwinConfig.local()
+        return MetaClient(config)
+
+    @classmethod
+    def from_api_key(cls, api_key: str, datasets_dir: Optional[Path] = None) -> Client:
```
Comment (owencjones) [inline darwin/future/meta/client.py:19]: Function sig says this returns client, but it returns `MetaClient` - which obv is a subclass anyway, but any reason not to just say `MetaClient` as a return?

File: darwin/future/meta/client.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+from requests.adapters import Retry
+
+from darwin.future.core.client import Client, DarwinConfig
+
+
+class MetaClient(Client):
+    def __init__(self, config: DarwinConfig, retries: Optional[Retry] = None) -> None:
+        super().__init__(config, retries=retries)
+
+    @classmethod
```
Comment (owencjones) [inline darwin/future/meta/client.py:13]: Same as other comment

File: darwin/future/meta/client.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    def local(cls) -> Client:
+        config = DarwinConfig.local()
+        return MetaClient(config)
```
Comment (owencjones) [inline darwin/future/meta/client.py:16]: And any reason these are `return MetaClient(config)` not `return cls(config)`?

**Review by owencjones — COMMENTED** [review]
Some queries, looking good

File: darwin/future/meta/client.py (hunk 2 above)
Comment (Nathanjp91) [reply darwin/future/meta/client.py]: good catch, I extracted this out from the core client so that's why.

**Review by owencjones — APPROVED** [review]

**Pull request merged**